firmware/library/L1_Drivers/pwm.hpp

```diff
@@ -1,13 +1,176 @@
 #pragma once
 
 #include <cstdint>
+#include <cstdio>
+#include "config.hpp"
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/pin_configure.hpp"
 
 class PwmInterface
 {
  public:
+    enum SysClk : uint8_t
+    {
+        kActivatePwm0 = (1 << 5),
+        kActivatePwm1 = (1 << 6)
+    };
+    enum PwmConfigure : uint8_t
+    {
+        kTimerMode = 0,
+        kUsePwmCap0 = (0 << 2),
+        kCounterEnable = (1 << 0),
+        kResetMr0 = (1 << 1),
+        kPwmEnable = (1 << 3)
+    };
```

Comment:
The two sections above aren't used by the interface methods. Not only that, but this interface is implementation specific to the LPC40xx boards. These shouldn't be apart of the interface. Remember that an interface is a contract for what ALL pwm classses must follow. If I make a PWM class using this interface for the ESP32, I won't have the same PWM configure register bits.

Please read: http://best-practice-software-engineering.ifs.tuwien.ac.at/patterns/interface.html